js/util/browser/dispatcher.js

```diff
@@ -17,9 +18,15 @@ function Dispatcher(length, parent) {
 }
 
 Dispatcher.prototype = {
-    broadcast: function(type, data) {
-        for (var i = 0; i < this.actors.length; i++) {
-            this.actors[i].send(type, data);
+    broadcast: function(type, data, cb) {
+        if (typeof cb !== 'function') {
+            for (var i = 0; i < this.actors.length; i++) {
+                this.actors[i].send(type, data);
+            }
+        } else {
+            util.asyncAll(this.actors, function (actor, done) {
+                actor.send(type, data, done);
+            }, cb);
```

Comment:
Can we remove the branching in this method by providing a default value for `cb` of `function() {}`?

```js
cb = cb || function() {};
```